packages/react/react/src/hooks/useUnmount.ts

```diff
@@ -0,0 +1,9 @@
+import { useEffect } from 'react';
+import { usePreservedCallback } from './usePreservedCallback';
+
+/** @tossdocs-ignore */
+export function useUnmount(callback: () => any){
+  const preservedCallback = usePreservedCallback(callback);
```

Comment:
By using usePreservedCallback, we maintain the reference and ensure it is up-to-date.